netsim/cli/api.py

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env python3
+#
+# netlab api command: lightweight HTTP wrapper for CLI actions
+#
+import argparse
+import base64
+import binascii
+import contextlib
+import datetime as dt
+import hmac
+import io
+import json
+import os
+import ssl
+import threading
+import time
+import traceback
+import urllib.parse
+from http import HTTPStatus
+from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer
+from typing import Any, Callable, Dict, List, Optional
+
+from ..utils import log
+from . import collect as netlab_collect
+from . import create as netlab_create
+from . import down as netlab_down
+from . import status as netlab_status
+from . import up as netlab_up
+
+DATA_DIR = os.getenv("NETLAB_API_DATA_DIR", "/var/lib/netlab/api")
```

Comment:
I don't think this is user-writeable on most systems, so the HTTP server would have to be started as root. I usually use "/tmp/something" in these cases (or you could use "~/...")